romancal/resample/resample.py

```diff
@@ -328,7 +334,13 @@ def resample_many_to_one(self):
                     and img.meta.background.subtracted is False
                     and img.meta.background.level is not None
                 ):
-                    data = img.data - img.meta.background.level
+                    if img.data.unit == img.meta.background.level.unit:
+                        data = img.data - img.meta.background.level
+                    else:
+                        data = img.data - (
+                            img.meta.background.level.value
+                            * img.meta.photometry.conversion_megajanskys
+                        ).to(img.data.unit)
```

Comment:
Are these bits needed to support unit tests or what?  I think in the real pipeline the first step in converting to MJy/sr so that sky matching should only ever see MJy/sr.  A case where sky matching sees DN / s and then we need to convert to MJy/sr seems like a likely bug.

Reply:
I believe unit tests are what spurred this code. I can remove this and update the tests, if preferred?

Reply:
Yes, ideally this code would just be
https://github.com/spacetelescope/romancal/pull/1351/files#diff-f87241f9a890ac31d6813379a09b6eb7e1052e3ab53039449a8f0df91aa030e1R251
and I worry that any unit changing is likely hiding bugs.  If it's easy to fix the tests so that they work with the right units, that would be great.

And it's not ideal to always work in DN / s---it feels like we should be working in MJy/sr.  But it's especially not ideal for this to be working in mixed units (e.g., sky subtraction in DN/s, but applied to MJy/sr), because it feels like in that case something is definitely screwed up and we're coding around that when in reality we should be raising an error.

Reply:
Code reverted and tests fixed.